Common/Securities/SecurityManager.cs

```diff
@@ -412,6 +432,46 @@ protected virtual void OnCollectionChanged(NotifyCollectionChangedEventArgs chan
         }
 
         /// <summary>
+        /// Get the data feed type for the given security
+        /// </summary>
+        /// <param name="factoryType"><see cref="BaseData"/> type of the security</param>
+        /// <param name="tickType">The <see cref="TickType"/> of the security</param>
+        /// <returns>Type that should be added as a subscription</returns>
+        private static Type GetDataFeedType(Type factoryType, TickType tickType)
+        {
+            Type dataFeedType;
+
+            // if it is a standard QC Data type
+            if (factoryType == typeof(TradeBar) ||
+                factoryType == typeof(QuoteBar) ||
+                factoryType == typeof(OpenInterest))
+            {
```

Comment:
This method is used to determine the `Type` field on the `SubscriptionDataConfig` and therefore the `BaseData.Reader()` used.  For 'non-common' security types (Bitcoin etc.) my understating is that even though the subscription has a resolution of Tick, the Reader used will still need to be the `Bitcoin.Reader()` and not `Tick.Reader()`.

I liked the idea of the LeanData method that checks if it is a 'common' QC data.  I added such a method to the LeanData class in #680.  This helped me refactor the `SecurityManager.GetDataFeedType()`.

Now, if the data is a common QC data type, a check for Resolution.Tick is run.  If we are dealing with a non-common QC data type then no such check is made.